Assets/Scripts/AnimalFactory.cs

```diff
@@ -0,0 +1,248 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using System.IO;
+
+public class AnimalFactory : MonoBehaviour
+{
+    public Texture2D formatTexture;
+    private string animalPrefabPath = "Prefab/Animals";
+    private string pngSavePath = "";
+
+    // ����� ���� �����͸� �޾Ƽ� �ٽ� ������ ���
+    [System.Serializable]
+    public class AnimalDataFromServer
+    {
+        public string name;
+        public int tier;
+        public string color;
+        public string id;
+        public string animalType;
+        public string headItem;
+        public string pattern;
+    }
+
+    void Start()
+    {
+        pngSavePath = $"{Application.dataPath}/RuntimeImages";        
+    }
+
+    public List<GameObject> ConvertJsonToAnimalObject(string jsonData)
+    {
+        List<GameObject> animalObjects = new List<GameObject>();
+
+        AnimalDataFromServer[] animalDatas = GetAnimalDataFromJson(jsonData);
+        foreach (AnimalDataFromServer data in animalDatas)
+        {
+            GameObject animalObj = GetAnimalGameObject(data);
+            animalObjects.Add(animalObj);
+        }
+
+        return animalObjects;
+    }
+
+    private AnimalDataFromServer[] GetAnimalDataFromJson(string txt)
+    {
+        var animalData = JsonHelper.FromJson<AnimalDataFromServer>(txt);
+        Debug.Log($"animalData = {animalData.Length}");
+
+        for(int i=0; i < animalData.Length; i++)
+        {
+            Debug.Log($"name = {animalData[i].name}, tier = {animalData[i].tier}, animalType = {animalData[i].animalType}");
+            Debug.Log($"color = {animalData[i].color}");
+            // Color ��� ���� �ؽ�ó ����
+            string color = animalData[i].color;
+            color = color.Replace("\\", string.Empty);
+
+            SaveJsonColorToTexture(color, animalData[i].animalType, animalData[i].id);
+        }              
+
+        return animalData;
+    }
+
+    GameObject GetAnimalGameObject(AnimalDataFromServer animalData)
+    {
+        Debug.Log($"type = {animalData.animalType}");
+        string path = $"{animalPrefabPath}/{animalData.animalType}";
+        GameObject animalPrefab = Resources.Load<GameObject>(path);
+
+        GameObject animalObj = GameObject.Instantiate(animalPrefab);
+        animalObj.transform.LookAt(Camera.main.transform);
+        animalObj.name = $"{animalData.animalType}_{animalData.id}";
+
+        // �ؽ�ó ����
+        Texture2D meshTex = GetTexture(animalData.animalType, animalData.id);        
+        animalObj.GetComponentInChildren<Renderer>().material.SetTexture("_MainTex", meshTex);
+
+        return animalObj;
+    }
+
+    private void SaveJsonColorToTexture(string colorJsonStr, string animalType, string id)
```

Comment:
I think, we create color class and add json2color, color2json, to save method will be better.
When we refactor code, we will see this line